core/src/main/java/com/flipkart/gjex/core/filter/Filter.java

```diff
@@ -63,4 +63,9 @@ public void doProcessResponse(Res response) {}
      * @param e The exception that occurred.
      */
     public void doHandleException(Exception e) {}
+
+    /**
+     * Method for performing ending functions for a filter.
+     */
+    public void doEndFilter() {}
```

Comment:
Not in favor of this. We already have an `destroy` method. Lifecycle methods should be used only for lifecycle and not for logic.

Reply:
This is not for ending lifecycle. This is for printing logs once only.